composer/workflows/s3togcsoperator_tutorial.py

```diff
@@ -0,0 +1,149 @@
+
+    s3_to_gcs_op = S3ToGCSOperator(
+        task_id="s3_to_gcs",
+        bucket=S3_BUCKET_NAME,
+        gcp_conn_id="google_cloud_default",
+        aws_conn_id="aws_s3_connection",
+        dest_gcs=f"gs://{BUCKET_NAME}",
+    )
+
+    create_batch = dataproc.DataprocCreateBatchOperator(
+        task_id="create_batch",
+        project_id=PROJECT_NAME,
+        region="{{ var.value.gce_region }}",
+        batch=BATCH_CONFIG,
+        batch_id=BATCH_ID,
+    )
+
+    load_external_dataset = GCSToBigQueryOperator(
+        task_id="run_bq_external_ingestion",
+        bucket=BUCKET_NAME,
+        source_objects=["holidays.csv"],
+        destination_project_dataset_table=f"{BQ_DESTINATION_DATASET_NAME}.holidays",
+        source_format="CSV",
+        schema_fields=[
+            {"name": "Date", "type": "DATE"},
+            {"name": "Holiday", "type": "STRING"},
+        ],
+        skip_leading_rows=1,
+        write_disposition="WRITE_TRUNCATE"
+    )
+
+    with TaskGroup("join_bq_datasets") as bq_join_group:
+
+        for year in range(1997, 2022):
+            # BigQuery configs
+            BQ_DATASET_NAME = f"bigquery-public-data.ghcn_d.ghcnd_{str(year)}"
+            BQ_DESTINATION_TABLE_NAME = "holidays_weather_joined"
+            # Specifically query a Chicago weather station
+            WEATHER_HOLIDAYS_JOIN_QUERY = f"""
+            SELECT Holidays.Date, Holiday, id, element, value
+            FROM `{PROJECT_NAME}.holiday_weather.holidays` AS Holidays
+            JOIN (SELECT id, date, element, value FROM {BQ_DATASET_NAME} AS Table WHERE Table.element="TMAX" AND Table.id="USW00094846") AS Weather
+            ON Holidays.Date = Weather.Date;
+            """
+
+            # for demo purposes we are using WRITE_APPEND
+            # but if you run the DAG repeatedly it will continue to append
+            # Your use case may be different, see the Job docs
+            # https://cloud.google.com/bigquery/docs/reference/rest/v2/Job
+            # for alternative values for the writeDisposition
+            # or consider using partitioned tables
+            # https://cloud.google.com/bigquery/docs/partitioned-tables
+            bq_join_holidays_weather_data = BigQueryInsertJobOperator(
+                task_id=f"bq_join_holidays_weather_data_{str(year)}",
+                configuration={
+                    "query": {
+                        "query": WEATHER_HOLIDAYS_JOIN_QUERY,
+                        "useLegacySql": False,
+                        "destinationTable": {
+                            "projectId": PROJECT_NAME,
+                            "datasetId": BQ_DESTINATION_DATASET_NAME,
+                            "tableId": BQ_DESTINATION_TABLE_NAME,
+                        },
+                        "writeDisposition": "WRITE_APPEND",
+                    }
+                },
+                location="US",
+            )
+
+        s3_to_gcs_op >> load_external_dataset >> bq_join_group >> create_batch
```

Comment:
Could you add a comment to highlight what the user might be able to expect? Will it print a result, update your DAG and be able to verify somewhere or...? Apologies if this is obvious as I'm not a Composer expert 😅

Reply:
Do you still want to see a comment if this will have an accompanying tutorial on c.g.c?

Reply:
It will have a region tag, we're just not at that stage of the program yet.